components/remote_settings/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pub mod error;
+use config::RemoteSettingsConfig;
+pub use error::{RemoteSettingsError, Result};
+mod config;
+use rs_client::{Client, ClientConfig};
+use serde::Deserialize;
+use std::{collections::HashMap, fs::File, io::prelude::Write};
+
+uniffi_macros::include_scaffolding!("remotesettings");
+
+pub struct RemoteSettings {
+    pub config: RemoteSettingsConfig,
```

Comment:
nit: does this need to be `pub`?

Reply:
I had imagined that it might be useful to be able to inspect the configuration at the application layer after construction, but I'm open to making it private if that isn't typical